Client/Internal/RetryAttempt.cs

```diff
@@ -82,25 +83,37 @@ internal bool IsRetry()
         /// <returns>retry interval to sleep</returns>
         internal long GetRetryInterval()
         {
-            long retryInterval;
-
             // from header
             if (Error is HttpException httpException && httpException.RetryAfter.HasValue)
             {
-                retryInterval = httpException.RetryAfter.Value * 1000;
+                return httpException.RetryAfter.Value * 1000 + JitterDelay(_writeOptions);
             }
+
             // from configuration
-            else
+            var rangeStart = _writeOptions.RetryInterval;
+            var rangeStop = _writeOptions.RetryInterval * _writeOptions.ExponentialBase;
+
+            var i = 1;
+            while (i < _count)
             {
-                retryInterval = _writeOptions.RetryInterval
-                                * (long) (Math.Pow(_writeOptions.ExponentialBase, _count - 1));
-                retryInterval = Math.Min(retryInterval, _writeOptions.MaxRetryDelay);
+                i++;
+                rangeStart = rangeStop;
+                rangeStop = rangeStop * _writeOptions.ExponentialBase;
+                if (rangeStop > _writeOptions.MaxRetryDelay)
+                {
+                    break;
+                }
+            }
 
-                Trace.WriteLine($"The InfluxDB does not specify \"Retry-After\". " +
-                                $"Use the default retryInterval: {retryInterval}");
+            if (rangeStop > _writeOptions.MaxRetryDelay)
+            {
+                rangeStop = _writeOptions.MaxRetryDelay;
             }
 
-            retryInterval += JitterDelay(_writeOptions);
+            var retryInterval = _random.Next(rangeStart, rangeStop);
+
+            Trace.WriteLine($"The InfluxDB does not specify \"Retry-After\". " +
```

Comment:
Redundant string interpolation:

```suggestion
            Trace.WriteLine("The InfluxDB does not specify \"Retry-After\". " +
```